main.go

```diff
@@ -0,0 +1,28 @@
+package main
```

Comment:
I had assumed we'd accept registrations on a separate port, and have the firewall restrict access to it.  But 2) is also a viable option.  Does anyone have a strong preference?